app/models/task.rb

```diff
@@ -20,4 +20,20 @@ class Task < ActiveRecord::Base
   validates :resource, presence: true
   validates :trial, presence: true,
                     uniqueness: { scope: [:identifier_id, :resource_id] }
+
+  # Look for a duplicate record in the DB. If not found, save and return self.
+  def find_or_create
+    # Try to find duplicates in the DB
+    duplicate_task = Task.find_by(identifier: identifier,
+                                  resource: resource,
+                                  trial: trial)
+
+    # Return pre-existing task if found
+    return duplicate_task unless duplicate_task.nil?
+
+    # No duplicates. Attempt to save and return self.
+    save
```

Comment:
This is currently handled by the `transfer_errors_from event.task` in the CreateEvent routine.

But if there are better ways to handle this, I'm open to suggestions.